plutus-ir/src/Language/PlutusIR/Parser.hs

```diff
@@ -0,0 +1,344 @@
+{-# LANGUAGE FlexibleContexts #-}
+module Language.PlutusIR.Parser where
+
+import           Prelude                      hiding (fail)
+
+import           Control.Applicative          hiding (many, some)
+import           Control.Monad.State          hiding (fail)
+
+import qualified Language.PlutusCore          as PLC
+import qualified Language.PlutusCore.Constant as PLC
+import           Language.PlutusIR            as PIR
+import           Text.Megaparsec              hiding (State, parse)
+
+import           Data.ByteString.Internal     (c2w)
+import qualified Data.ByteString.Lazy         as BS
+import           Data.Char
+import           Data.Foldable
+import qualified Data.Map                     as M
+import qualified Data.Text                    as T
+import           Data.Word
+import           GHC.Natural
+
+import           Text.Megaparsec.Char
+import qualified Text.Megaparsec.Char.Lexer   as Lex
+
+data ParserState = ParserState { identifiers :: M.Map String PLC.Unique, counter :: PLC.Unique }
+    deriving (Show)
+
+data PIRParseError = Overflow Natural Integer
```

Comment:
Prefer not to prefix things with namespacing information - we have qualified imports for that. `ParseError` is fine.

Reply:
Ah I see it clashes with the megaparsec `ParseError`. I'd probably import that qualified tbh.